resolwe/flow/views.py

```diff
@@ -456,20 +456,24 @@ class CollectionViewSet(ResolweCreateModelMixin,
     permission_classes = (permissions_cls,)
     filter_class = CollectionFilter
 
+    def _check_object_permissions(self, collection, user):
```

Comment:
Naming is intentional, because you have to check the permissions on Sample in SampleViewSet and you just overwrite this function.